components/validator/l2_output_submitter.go

```diff
@@ -202,8 +223,27 @@ func (l *L2OutputSubmitter) canSubmit(ctx context.Context) (*big.Int, bool, erro
 	return nextBlockNumber, true, nil
 }
 
-func (l *L2OutputSubmitter) fetchBlockNumbers(ctx context.Context) (*big.Int, *big.Int, error) {
-	cCtx, cCancel := context.WithTimeout(ctx, l.cfg.NetworkTimeout)
+func (l *L2OutputSubmitter) checkDeposit() (bool, error) {
+	cCtx, cCancel := context.WithTimeout(l.ctx, l.cfg.NetworkTimeout)
+	defer cCancel()
+	from := l.cfg.TxManager.From()
+	callOpts := utils.NewCallOptsWithSender(cCtx, from)
+	balance, err := l.valpoolContract.BalanceOf(callOpts, from)
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch validator deposit amount: %w", err)
+	}
+
+	if balance.Cmp(l.minBondAmount) == -1 {
```

Comment:
The amount actually needed is the bondAmount in the config, not the `minBondAmount`.